AI/Nullkiller/Pathfinding/AINodeStorage.cpp

```diff
@@ -823,13 +811,6 @@ ExchangeCandidate HeroChainCalculationTask::calculateExchange(
 	return candidate;
 }
 
-const CGHeroInstance * AINodeStorage::getHero(const CGPathNode * node) const
-{
-	auto aiNode = getAINode(node);
-
-	return aiNode->actor->hero;
```

Comment:
Moved them to header to allow inline optimization. For some reason aiNode->actor sometimes appears to be null  while in crashdump memory it is perfectly valid. I can not understand what is going on but inlining seems to help somehow.

Reply:
so you can add inline keyword

Reply:
surprisingly enough, in c++ "inline" does not means that function should be inlined by compiler.
"inline" effectively means that this function is defined in header and as result - can appear in multiple compilation units.
And methods defined in place are implicitly inline'd. So no need to add inline keyword here.

Reply:
Yes, it's recommendation for compiler not obligation, but it would help developers to understand intention of having those methods in header

Reply:
Thing is - it is not recommendation. I remember having linker error because of incorrectly placed inline.
Won't matter in this particular case though.